### ISSUE TITLE
SlotFilling no longer works in V4 or documentation is incorrect

### ISSUE THREAD

**Issue by MattRiddell** [axa-group/nlp.js#702]
In the documentation on the home page slot filling is linked to but goes to a document in the v3 folder:

[https://github.com/axa-group/nlp.js/blob/master/docs/v3/slot-filling.md](https://github.com/axa-group/nlp.js/blob/master/docs/v3/slot-filling.md)

NlpManager no longer contains `slotManager` or `addTrimEntity`?

More specifically it says this:

```JavaScript
  const manager = new NlpManager({ languages: ['en'] });
  const fromEntity = manager.addTrimEntity('fromCity');
  fromEntity.addBetweenCondition('en', 'from', 'to');
  fromEntity.addAfterLastCondition('en', 'from');
  const toEntity = manager.addTrimEntity('toCity');
  toEntity.addBetweenCondition('en', 'to', 'from', { skip: ['travel'] });
  toEntity.addAfterLastCondition('en', 'to');
 
  manager.slotManager.addSlot('travel', 'fromCity', true, { en: 'From where you are traveling?' });
  manager.slotManager.addSlot('travel', 'toCity', true, { en: 'Where do you want to go?' });
  manager.slotManager.addSlot('travel', 'date', true, { en: 'When do you want to travel?' });


  manager.addDocument('en', 'I want to travel from %fromCity% to %toCity% %date%', 'travel');
  await manager.train();
  const result = await manager.process('en', 'I want to travel to Madrid tomorrow', {});
  console.log(JSON.stringify(result, null, 2));
```
The problem is that the slot manager stuff doesn't seem to be in NLPManager anymore so I'm guessing the code has to change?

I saw that by adding %name% in greetings.hello intent such as hello my name is %name% it does get added to the slot manager when I train it, but it doesn't appear to get parsed when I feed it "hello my name is matt".

This is also discussed in #561

**Comment by MattRiddell** [comment]
Also, if I can understand how this should work I can help update documentation.

I see the code has been split out but don't see how you would combine the slots manager into the above example.

**Comment by Apollon77** [comment]
@MattRiddell Did you manage to get slot filling somehow working with v4?

**Comment by Apollon77** [comment]
oook ... I got it working ...  maybe I find time to post a PR for documentation soon.

In fact all the methods have changed and are now on "nlp" class.

e.g. 
* entity.addBetweenCondition() -> nlp.addNerBetweenCondition()
* entity.addAfterLastCondition() -> nlp.addNerAfterLastCondition()

And the entities are automatically registered via addDocument, so you just add them using the above methods.

nlp.slotManager.addSlot stays as before.

I just wonder why all these Ner Rules can not also being parsed out of the utterances from addDocument because also here we can see words before, after, in between and such ... nbut I assume this would be a new feature request :-)

**Comment by Apollon77** [comment]
new examples and documentation and fixes will be in #1171 and other PRs
